.github/workflows/ci-detox-ios.yml

```diff
@@ -1,104 +1,135 @@
-#           override: true
-
-#       - name: Build Libraries for iOS
-#         run: yarn build:ios
-
-#       - name: Archive build artifacts
-#         if: success()
-#         uses: actions/upload-artifact@v3
-#         with:
-#           name: ci-detox-ios-artifacts-libs-${{ github.sha }}
-#           path: wrappers/react-native/ios/lib
-#           retention-days: 1
-
-#   ios:
-#     name: Detox iOS E2E Testing
-#     needs: [libraries]
-#     runs-on: macos-latest
-#     # strategy:
-#     #   matrix:
-#     #     runners: [[self-hosted, macos, general, detox-ios]]
-#     # concurrency:
-#     #   group: ci-detox-ios-${{ matrix.runners }}-${{ github.head_ref || github.run_id  }}
-#     #   cancel-in-progress: true
-#     timeout-minutes: 120
-
-#     defaults:
-#       run:
-#         working-directory: wrappers/react-native
-#         shell: zsh -il {0} # load ~/.zshrc
-
-#     env:
-#       DEVELOPER_DIR: /Applications/Xcode_14.2.app
-#       NODE_OPTIONS: "--max-old-space-size=5120"
-#       # Do not launch packager during CI build
-#       RCT_NO_LAUNCH_PACKAGER: 'true'
-
-#     steps:
-#       - name: Checkout
-#         uses: actions/checkout@v2
-
-#       - name: Setup Node
-#         uses: actions/setup-node@v3
-#         with:
-#           node-version: 18.x
-#           registry-url: https://registry.npmjs.org
-
-#       - name: Download iOS Libraries
-#         uses: actions/download-artifact@v3
-#         with:
-#           name: ci-detox-ios-artifacts-libs-${{ github.sha }}
-#           path: wrappers/react-native/ios/lib
-
-#       - name: Inspect iOS Libraries
-#         run: ls -R ios/lib
-
-#       - name: Install Dependencies
-#         run: yarn install --frozen-lockfile
-
-#       - name: Example App Install Dependencies
-#         run: yarn example --frozen-lockfile
-
-#       - name: Example App Install Pods
-#         run: yarn example pods
-
-#       - run: brew tap wix/brew
-#       - run: brew install applesimutils
-
-#       - name: Detox Build iOS Configuration
-#         run: yarn detox:build:ios
-
-#       - name: Detox E2E Test
-#         run: yarn detox:ios:ci
-
-#       - name: Archive detox failed test results
-#         if: failure()
-#         uses: actions/upload-artifact@v3
-#         with:
-#           name: detox-artifacts-${{ github.sha }}
-#           path: wrappers/react-native/example/detox-artifacts
-#           retention-days: 1
+name: ci-detox-ios
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  libraries:
+    name: Build Wrapper Libraries
+    runs-on: macos-14
+    defaults:
+      run:
+        working-directory: ./wrappers/react-native
+    steps:
+      - name: Checkout
+        uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # pin@v3.5.1
+
+      - name: Setup rust toolchain
+        uses: actions-rs/toolchain@b2417cde72dcf67f306c0ae8e0828a81bf0b189f # pin@v1.0.6
+        with:
+          profile: minimal
+          toolchain: 1.70.0
```

Comment:
Had to pin the toolchain otherwise I get missing symbols for ARM 👇. For now just going to pin but we should fix this so attaching details here for reference. 

```bash
ld: warning: object file (/Users/runner/work/pairing_crypto/pairing_crypto/target/aarch64-apple-ios/release/deps/libblst-6753210bfcbfe50f.rlib[16](f6e817b043e0335b-assembly.o)) was built for newer 'iOS' version (17.2) than being linked (10.0)
          ld: Undefined symbols:
            ___chkstk_darwin, referenced from:
                _POINTonE1_mult_glv in libblst-6753210bfcbfe50f.rlib[15](3ce72ea41a6346fd-server.o)
                _POINTonE2_mult_gls in libblst-6753210bfcbfe50f.rlib[15](3ce72ea41a6346fd-server.o)
                _POINTonE2_mult_w5 in libblst-6753210bfcbfe50f.rlib[15](3ce72ea41a6346fd-server.o)
                _HKDF_Expand in libblst-6753210bfcbfe50f.rlib[15](3ce72ea41a6346fd-server.o)
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Think we need to update `libblst` to resolve this issue or one of our wrappers. We also need to upgrade our ios project `obj-c` & `react-native` wrapper to make it eaiser to target newer IOS versions.

This isn't a direct dependency we have forked [blstrs](https://github.com/filecoin-project/blstrs) written in Rust and they have forked [blst](https://github.com/supranational/blst) written in C.

```bash
cargo tree | grep blst
├── blstrs v0.6.1 (https://github.com/mattrglobal/blstrs.git?rev=a0cb960#a0cb9601)
│   ├── blst v0.3.10
│   ├── blstrs v0.6.1 (https://github.com/mattrglobal/blstrs.git?rev=a0cb960#a0cb9601) (*) 
```

Reply:
At a glance, it [looks like it has problems loading chkstk_darwin](https://github.com/realm/realm-swift/issues/7586) if you have a library built with iOS 13 target. An iOS 12 device will fail to link the symbol at runtime. So yes the fix would be to update iOS I suppose! Though, this looks like something that could cause compatibility problem on older devices, and we may have obligations to support those

Reply:
Yeah that's right 👍, we will need to update the base project don't want to do that here.

> cause compatibility problem on older devices

Yeah that will be for others to decide but eventually some devices will have to get cut.

It's only if that particular device cannot upgrade to that minimum version when it becomes an issue with some grace period of course. Would have to look at market share to figure it out.